text/0000-components.md

```diff
@@ -0,0 +1,261 @@
+
+The Grid CLI should feel natural on a Linux system and adhere to Linux CLI
+norms and should be suitable for use when scripting. For example, when an error
+occurs, the CLI command should exit with a non-zero exit status so that it can
+be checked in a bash script. The command’s output should be both human friendly
+and easily parsable with common Linux command line tools such as awk and sed
+(when necessary, as different command-line options). 
+
+The Grid CLI will use the Rust clap library to parse command line arguments,
+with restrictions. Problematic multiple-value syntax such as `-o <val1> <val2>`
+should be avoided and `-o <val1> -o <val2>` form used instead.
+
+## SDK
+
+The Grid SDK provides programming support for application developers and
+systems integrators. Initially, only the Rust programming language is
+supported, though others may be proposed in future RFCs.
+
+In particular, the SDK includes APIs and implementations to help write smart
+contracts, including:
+
+- Reading product information from state
+- Reading Pike identity information from state
+- Reading schema information from state
+- Pike permission verifier
+- Schema compliance verify
+
+Like the other components, the SDK will grow over time as new features are
+supported by Grid. When adding new features or making changes to the SDK,
+backwards compatibility should be considered. The SDK’s versioning should
+conform to semver (which is true of all components, but of particular
+importance for the SDK).
+Smart Contracts
+
+Several features of Hyperledger Grid use smart contracts as part of their
+implementation. The smart contracts are deployed in Sabre, which is part of
+Hyperledger Sawtooth. Sabre executes smart contracts which have been compiled
+to WebAssembly. Most of the smart contracts which are formally part of
+Hyperledger Grid will be written in the Rust programming language. However,
+future RFCs may extend this to additional programming languages supported by
+future versions of Sabre.
+
+Some initial smart contracts supported by Grid are:
+
+- Grid Product
+- Grid Pike
+- Grid Track and Trace
+- Grid Schemas
+
+Smart contracts added to Grid will likely always be accompanied by an RFC which
+documents the problem they solve and technical details needed to write and
+maintain the smart contract.
+
+## Documentation
+
+Hyperledger Grid’s documentation plays a critical role in describing Grid to
+potential users and providing necessary guide and reference material for
+application developers, systems integrators, and systems administrators.
+
+The team feels strongly that “documentation is source code” and the approach to
+maintaining it matches the processes we use for all other development. As such,
+we have chosen formats for documentation which are friendly to source control
+under git while providing the best output possible. The documentation is
+written in: a) Rust doc comments for SDK API documentation; b) sphinx-doc for
```

Comment:
s/sphinx-doc/ReStructuredText (RST) with Sphinx/

... because the markup language is RST, and the doc-generation tool is Sphinx (not sphinx-doc - that's the web domain).